tests/src/image.rs

```diff
@@ -505,24 +550,70 @@ impl ReadbackBuffers {
         copy_texture_to_buffer(device, encoder, texture, &self.buffer, &self.buffer_stencil);
     }
 
+    fn retrieve_buffer(
+        &self,
+        device: &Device,
+        buffer: &Buffer,
+        aspect: Option<TextureAspect>,
+    ) -> Vec<u8> {
+        let buffer_slice = buffer.slice(..);
+        buffer_slice.map_async(MapMode::Read, |_| ());
+        device.poll(Maintain::Wait);
+        let (block_width, block_height) = self.texture_format.block_dimensions();
+        let expected_bytes_per_row = (self.texture_width / block_width)
+            * self.texture_format.block_size(aspect).unwrap_or(4);
+        let expected_buffer_size = expected_bytes_per_row
+            * (self.texture_height / block_height)
+            * self.texture_depth_or_array_layers;
+        let data: BufferView = buffer_slice.get_mapped_range();
+        if expected_buffer_size as usize == data.len() {
+            data.to_vec()
+        } else {
+            bytemuck::cast_slice(&data)
+                .to_vec()
+                .chunks_exact(
+                    align_to(expected_bytes_per_row, COPY_BYTES_PER_ROW_ALIGNMENT) as usize,
+                )
+                .flat_map(|x| x.iter().take(expected_bytes_per_row as usize))
+                .copied()
+                .collect()
+        }
+    }
+
+    fn buffer_aspect(&self) -> Option<TextureAspect> {
+        if self.texture_format.is_combined_depth_stencil_format() {
+            Some(TextureAspect::DepthOnly)
+        } else {
+            None
+        }
+    }
+
     pub fn are_zero(&self, device: &Device) -> bool {
-        fn is_zero(device: &Device, buffer: &Buffer) -> bool {
-            let is_zero = {
-                let buffer_slice = buffer.slice(..);
-                buffer_slice.map_async(MapMode::Read, |_| ());
-                device.poll(Maintain::Wait);
-                let buffer_view = buffer_slice.get_mapped_range();
-                buffer_view.iter().all(|b| *b == 0)
-            };
+        let is_zero = |device: &Device, buffer: &Buffer, aspect: Option<TextureAspect>| -> bool {
+            let is_zero = self
+                .retrieve_buffer(device, buffer, aspect)
+                .iter()
+                .all(|b| *b == 0);
             buffer.unmap();
             is_zero
-        }
+        };
 
-        is_zero(device, &self.buffer)
+        is_zero(device, &self.buffer, self.buffer_aspect())
             && self
                 .buffer_stencil
                 .as_ref()
-                .map(|buffer_stencil| is_zero(device, buffer_stencil))
+                .map(|buffer_stencil| {
+                    is_zero(device, buffer_stencil, Some(TextureAspect::StencilOnly))
+                })
```

Comment:
Could we split this into two bools which are then and'ed

Reply:
Honestly, I prefer the way that it was written, but I agree that at first, it is a little bit hard to grasp this logic.
refactored this one, too